volatility3/framework/plugins/linux/sockstat.py

```diff
@@ -0,0 +1,374 @@
+            saddr_list.append(groups_str)
+
+        daddr_list = []
+        dst_portid = f"portid:{netlink_sock.dst_portid}"
+        daddr_list.append(dst_portid)
+        dst_group = f"group:0x{netlink_sock.dst_group:08x}"
+        daddr_list.append(dst_group)
+        module = netlink_sock.module
+        if module and netlink_sock.module.name:
+            module_name_str = utility.array_to_string(netlink_sock.module.name)
+            module_name = f"lkm:{module_name_str}"
+            daddr_list.append(module_name)
+
+        saddr_tag = ",".join(saddr_list)
+        daddr_tag = ",".join(daddr_list)
+        state = netlink_sock.state
+
+        sock_stat = saddr_tag, daddr_tag, state
+        return netlink_sock, sock_stat
+
+    def _vsock_sock(self, sock, _extended):
+        vsock_sock = sock.cast("vsock_sock")
+        saddr = vsock_sock.local_addr.svm_cid
+        sport = vsock_sock.local_addr.svm_port
+        daddr = vsock_sock.remote_addr.svm_cid
+        dport = vsock_sock.remote_addr.svm_port
+        state = ""  # Protocol is always 0
+
+        saddr_tag = f"{saddr}:{sport}"
+        daddr_tag = f"{daddr}:{dport}"
+        sock_stat = saddr_tag, daddr_tag, state
+        return vsock_sock, sock_stat
+
+    def _packet_sock(self, sock, extended):
+        packet_sock = sock.cast("packet_sock")
+        ifindex = packet_sock.ifindex
+        dev_name = self._netdevices.get(ifindex, "") if ifindex > 0 else "ANY"
+
+        if sock.has_member("sk_filter"):
+            sock_filter = sock.sk_filter
+            self.__update_extra_socket_bpf(sock_filter, extended)
+
+        if sock.has_member("sk_reuseport_cb"):
+            sock_reuseport_cb = sock.sk_reuseport_cb
+            self.__update_extra_socket_bpf(sock_reuseport_cb, extended)
+
+        saddr_tag = f"{dev_name}"
+        daddr_tag = ""
+        state = packet_sock.state
+        sock_stat = saddr_tag, daddr_tag, state
+        return packet_sock, sock_stat
+
+    def __update_extra_socket_bpf(self, sock_filter, extended):
+        if not sock_filter:
+            return
+
+        extended["bpf_filter_type"] = "cBPF"
+
+        if not sock_filter.has_member("prog"):
+            return
+
+        bpfprog = sock_filter.prog
+        if not bpfprog:
+            return
+
+        BPF_PROG_TYPE_UNSPEC = 0
+        if bpfprog.type > BPF_PROG_TYPE_UNSPEC:
+            extended["bpf_filter_type"] = "eBPF"
+            bpfprog_aux = bpfprog.aux
+            if bpfprog_aux:
+                extended["bpf_filter_id"] = str(bpfprog_aux.id)
+                bpfprog_name = str(utility.array_to_string(bpfprog.aux.name))
+                if bpfprog_name:
+                    extended["bpf_filter_name"] = bpfprog_name
+
+    def _xdp_sock(self, sock, _extended):
+        xdp_sock = sock.cast("xdp_sock")
+        device = xdp_sock.dev
+        if not device:
+            return
+
+        dev_name = utility.array_to_string(device.name)
+        saddr_tag = f"{dev_name}"
```

Comment:
This seems a little redundant?  It's not a show stopper, it's just making me scratch my head and wonder why it's there?

Reply:
Right. I thought array_to_string() returned an object other than str().
Is it the return type correct [here](https://github.com/volatilityfoundation/volatility3/blob/280c91212140ceb320ee59c647378e28f919592e/volatility3/framework/objects/utility.py#L12)? I think it was probably copy & pasted from _array_to_pointers_. Also _pointer_to_string_ needs a return type.

Reply:
It returns a `String` volatility object (so the return type is correct), but volatility `Strings` should be inherited from `str` and therefore shouldn't need replacing.  I might be wrong, in which case this is no problem, but I believe it's unnecessary, and if so I'd prefer it be taken out please...  5:)

Reply:
It was already changed same days ago. These changes are outdated.